src/plugins/visualizations/public/wizard/search_selection/search_selection.tsx

```diff
@@ -54,6 +56,7 @@ export class SearchSelection extends React.Component<SearchSelectionProps> {
           </EuiModalHeaderTitle>
         </EuiModalHeader>
         <EuiModalBody>
+          <DialogNavigation goBack={() => this.props.goBack()} />
```

Comment:
nit:  I think we can pass `goBack` props as goBack={ this.props.goBack } to don't re-render `DialogNavigation` on each time when parent props was changed